File: leet_remove_adj_dup_i_1047.js
/**
 * @param {string} s
 * @return {string}
 */

var removeDuplicates = function(s) {
    q = []
    for (let c of s) {
        if (q.length != 0 && c == q[0])
            q.shift()
        else
            q.unshift(c)
    }
    return q.reverse().join('')
};

File: leet_remove_adj_dup_i_1047.py
class Solution:
    def removeDuplicates(self, s: str) -> str:
        q = []
        for c in s:
            if len(q) != 0 and c == q[-1]:
                q.pop()
            else:
                q.append(c)
        return ''.join(q)
